File: Day 56/Solution.py
# 9. Palindrome Number
# Given an integer x, return true if x is a palindrome, and false otherwise.
from typing import Optional

class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        
        num = str(x)
        l, r = 0, len(num) - 1
        
        while l < r:
            if num[l] != num[r]:
                return False
            l += 1
            r -= 1
        return True

# Test block for VSCode
if __name__ == "__main__":
    s = Solution()
    print(s.isPalindrome(121))    # True
    print(s.isPalindrome(-121))   # False
    print(s.isPalindrome(10))     # False

File: Day 56/Solution.java
// 9. Palindrome Number
// Given an integer x, return true if x is a palindrome, and false otherwise.
public class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }
        
        String num = String.valueOf(x);
        int l = 0;
        int r = num.length() - 1;
        
        while (l < r) {
            if (num.charAt(l) != num.charAt(r)) {
                return false;
            }
            l++;
            r--;
        }
        return true;
    }

    // Test block for VSCode
    public static void main(String[] args) {
        Solution s = new Solution();
        System.out.println(s.isPalindrome(121));  // true
        System.out.println(s.isPalindrome(-121)); // false
        System.out.println(s.isPalindrome(10));   // false
    }
}
